velox/serializers/PrestoSerializer.cpp

```diff
@@ -4359,6 +4359,27 @@ void readTopColumns(
   readColumns(
       &source, childTypes, resultOffset, nullptr, 0, pool, opts, children);
 }
+
+std::unique_ptr<folly::IOBuf> uncompressStream(
+    ByteInputStream* source,
+    const PrestoHeader& header,
+    common::CompressionKind compressionKind,
+    memory::MemoryPool& pool) {
+  const auto codec = common::compressionKindToCodec(compressionKind);
+  if (dynamic_cast<BufferInputStream*>(source)) {
```

Comment:
Just do the `dynamic_cast` once: `if (const auto* bufferSource = dynamic_cast<BufferInputStream*>(source))`

Reply:
I implement the readBytes return IOBuf in FileInputStream, so we don't need to do the cast, and other compression code can also benefit from this.